src/tests/test_ucc_verifier.cpp

```diff
@@ -0,0 +1,153 @@
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "algorithms/ucc/ucc_verifier/ucc_verifier.h"
+#include "config/indices/type.h"
+#include "datasets.h"
+
+namespace tests {
+namespace onam = config::names;
+
+class UCCVerifierSimpleParams {
+private:
+    algos::StdParamsMap params_map_;
+    size_t num_clusters_violating_ucc_ = 0;
+    size_t num_rows_violating_ucc_ = 0;
+    std::vector<model::PLI::Cluster> clusters_violating_ucc_;
+
+public:
+    UCCVerifierSimpleParams(config::IndicesType column_indices,
+                            size_t const num_clusters_violating_ucc,
+                            size_t const num_rows_violating_ucc,
+                            std::vector<model::PLI::Cluster> clusters_violating_ucc,
+                            std::string_view dataset, char const separator = ',',
+                            bool const has_header = true)
+        : params_map_({{onam::kCsvPath, test_data_dir / dataset},
+                       {onam::kSeparator, separator},
+                       {onam::kHasHeader, has_header},
+                       {onam::kEqualNulls, true}}),
+          num_clusters_violating_ucc_(num_clusters_violating_ucc),
+          num_rows_violating_ucc_(num_rows_violating_ucc),
+          clusters_violating_ucc_(std::move(clusters_violating_ucc)) {
+        // if column_indices is empty then it is not inserted into params_map_ (and this option is
+        // not passed to the algorithm), in this case default implementation will be used
+        if (!column_indices.empty()) {
+            params_map_.insert({onam::kUCCIndices, std::move(column_indices)});
+        }
+    }
+
+    algos::StdParamsMap GetParamsMap() const {
+        return params_map_;
+    }
+
+    size_t GetExpectedNumClustersViolatingUCC() const {
+        return num_clusters_violating_ucc_;
+    }
+
+    size_t GetExpectedNumRowsViolatingUCC() const {
+        return num_rows_violating_ucc_;
+    }
+
+    std::vector<model::PLI::Cluster> const& GetExpectedClustersViolatingUCC() const {
+        return clusters_violating_ucc_;
+    }
+};
+
+class TestUCCVerifierSimple : public ::testing::TestWithParam<UCCVerifierSimpleParams> {};
+
+class UCCVerifierWithHyUCCParams {
```

Comment:
I'd move this class below the test for `TestUCCVerifierSimple` fixture, I think it will be easier to read that way, because fixture definition will be close to its usage